File: src/2038. Remove Colored Pieces if Both Neighbors are the Same Color.py
class Solution:
    def winnerOfGame(self, colors: str) -> bool:
        alice, bob = 0, 0
        for i in range(2, len(colors)):
            if colors[i - 2] == colors[i - 1] == colors[i]:
                if colors[i] == 'A':
                    alice += 1
                else:
                    bob += 1
        return alice > bob
            

File: src/2038. Remove Colored Pieces if Both Neighbors are the Same Color.cpp
class Solution {
public:
    bool winnerOfGame(string colors) {
        int alice = 0, bob = 0;
        for (int i = 2; i < colors.size(); ++i) {
            if (colors[i - 2] == colors[i - 1] && colors[i - 1] == colors[i]) {
                if (colors[i] == 'A') {
                    alice++;
                } else {
                    bob++;
                }
            }
        }
        return alice > bob;
    }
};